rotkehlchen/globaldb/handler.py

```diff
@@ -767,11 +763,17 @@ def check_asset_exists(
         If it exists it returns a list of the identifiers of the assets.
         """
         cursor = GlobalDBHandler().conn.cursor()
-        query = cursor.execute(
-            'SELECT A.identifier from assets AS A JOIN common_asset_details as C '
-            'ON A.identifier=C.identifier WHERE A.type=? AND A.name=? AND C.symbol=?;',
-            (asset_type.serialize_for_db(), name, symbol),
-        )
+        if isinstance(asset, EvmToken):
+            query = cursor.execute(
+                'SELECT identifier FROM evm_tokens WHERE identifier=?',
+                (asset.identifier,),
+            )
+        else:
+            query = cursor.execute(
+                'SELECT A.identifier from assets AS A JOIN common_asset_details as C '
```

Comment:
Hmmm shouldn't this also check by ID? I don't get why check name/symbol here. What if there is multiple assets with same name and symol and type?

Reply:
Answered in discord. Is because for evm tokens we know 100% by the address/chain/token kind if the asset exists just by checking the identifier. For other assets is not possible to do so with the user provided by the user